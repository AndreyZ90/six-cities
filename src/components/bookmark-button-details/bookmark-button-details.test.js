import React from 'react';
import TestRenderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore([]);

const store = mockStore({
  user: {
    authStatus: `AUTH`
  }
});

import BookmarkButtonDetails from '@/components/bookmark-button-details/bookmark-button-details';

describe(`BookmarkButtonDetails component snapshot`, () => {
  test(`Should correctly render BookmarkButtonDetails component (active)`, () => {
    const tree = TestRenderer.create(
        <Provider store={store}>
          <Router>
            <BookmarkButtonDetails isActive={true} id={10} />
          </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render BookmarkButtonDetails component (not active)`, () => {
    const tree = TestRenderer.create(
        <Provider store={store}>
          <Router>
            <BookmarkButtonDetails isActive={false} id={10} />
          </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
