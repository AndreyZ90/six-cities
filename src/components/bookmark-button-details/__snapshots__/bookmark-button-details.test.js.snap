// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkButtonDetails component snapshot Should correctly render BookmarkButtonDetails component (active) 1`] = `
<button
  className="property__bookmark-button button property__bookmark-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height={33}
    width={31}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`BookmarkButtonDetails component snapshot Should correctly render BookmarkButtonDetails component (not active) 1`] = `
<button
  className="property__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height={33}
    width={31}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
